x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/index.tsx

```diff
@@ -0,0 +1,415 @@
+
+    .unifiedFieldListSidebar__group {
+      .euiFlexItem:last-child {
+        /* padding-right: ${(props) => (props.theme as EuiTheme).eui.euiSizeS}; */
+      }
+      .unifiedFieldListSidebar__list {
+        padding-left: 0px;
+      }
+
+      .unifiedFieldListSidebar__addBtn {
+        margin-right: ${(props) => (props.theme as EuiTheme).eui.euiSizeS};
+      }
+    }
+  }
+`;
+
+export const SAMPLE_SIZE_SETTING = 500;
+export const HIDE_FOR_SIZES = ['xs', 's'];
+
+interface Props {
+  columns: ColumnHeaderOptions[];
+  rowRenderers: RowRenderer[];
+  timelineId: string;
+  itemsPerPage: number;
+  itemsPerPageOptions: number[];
+  sort: SortColumnTimeline[];
+  events: TimelineItem[];
+  refetch: inputsModel.Refetch;
+  totalCount: number;
+  onEventClosed: (args: ToggleDetailPanel) => void;
+  expandedDetail: ExpandedDetailTimeline;
+  showExpandedDetails: boolean;
+  onChangePage: OnChangePage;
+  activeTab: TimelineTabs;
+  dataLoadingState: DataLoadingState;
+  updatedAt: number;
+  isTextBasedQuery?: boolean;
+}
+
+export const UnifiedTimelineComponent: React.FC<Props> = ({
+  columns,
+  activeTab,
+  timelineId,
+  itemsPerPage,
+  itemsPerPageOptions,
+  rowRenderers,
+  sort,
+  events,
+  refetch,
+  dataLoadingState,
+  totalCount,
+  onEventClosed,
+  showExpandedDetails,
+  expandedDetail,
+  onChangePage,
+  updatedAt,
+  isTextBasedQuery,
+}) => {
+  const dispatch = useDispatch();
+  const unifiedFieldListContainerRef = useRef<UnifiedFieldListSidebarContainerApi>(null);
+  const renderId = useRef(1);
+
+  const {
+    services: {
+      uiSettings,
+      fieldFormats,
+      dataViews,
+      dataViewFieldEditor,
+      application: { capabilities },
+      data: dataPluginContract,
+      uiActions,
+      charts,
+      docLinks,
+      analytics,
+    },
+  } = useKibana();
+
+  const fieldListSidebarServices: UnifiedFieldListSidebarContainerProps['services'] = useMemo(
+    () => ({
+      fieldFormats,
+      dataViews,
+      dataViewFieldEditor,
+      data: dataPluginContract,
+      uiActions,
+      charts,
+      core: {
+        analytics,
+        uiSettings,
+        docLinks,
+      } as CoreStart,
+    }),
+    [
+      fieldFormats,
+      dataViews,
+      dataViewFieldEditor,
+      dataPluginContract,
+      uiActions,
+      charts,
+      uiSettings,
+      docLinks,
+      analytics,
+    ]
+  );
+
+  const [sidebarContainer, setSidebarContainer] = useState<HTMLDivElement | null>(null);
+  const [, setMainContainer] = useState<HTMLDivElement | null>(null);
+
+  const { timeline: { filterManager } = timelineDefaults } = useSelector((state: State) =>
+    timelineBodySelector(state, timelineId)
+  );
+
+  const defaultColumns = useMemo(() => {
+    return columns.map((c) => c.id);
+  }, [columns]);
+  const { sourcererDataView } = useSourcererDataView(SourcererScopeName.timeline);
+
+  const dataView = useMemo(() => {
```

Comment:
Could we use the `use_get_sourcerer_data_view` hook here?

Reply:
It is done here: 1fe3ea63eb0ad135e21cbf67e03d8af55cad0574

Reply:
Thanks!